### DISCUSSION TITLE
Does `uv.run()` block current nvim editor? Or it runs in background (just like in another thread or something)?

### DISCUSSION THREAD

**Discussion by linrongbin16** [neovim/neovim#24641] [Q&A]
Hi, based on this doc: https://neovim.io/doc/user/luvref.html#luv-intro, if I setup a TCP server follow the sample code:

```lua
local uv = vim.loop
local server = uv.new_tcp()
server:bind("127.0.0.1", 1337)
server:listen(128, function (err)
  assert(not err, err)
  local client = uv.new_tcp()
  server:accept(client)
  client:read_start(function (err, chunk)
    assert(not err, err)
    if chunk then
      client:write(chunk)
    else
      client:shutdown()
      client:close()
    end
  end)
end)
print("TCP server listening at 127.0.0.1 port 1337")
uv.run() -- an explicit run call is necessary outside of luvit
```

Will `uv.run()` in the last line block the nvim editor??? Or it always run in background???

**Comment by bfredl** [comment]
That example should be adapted by removing `uv.run()` , as there is no need to call it for lua code which runs in nvim's main thread. (The event loop will run by nvim whenever nvim needs to check for input)
